projects/storefrontlib/layout/a11y/keyboard-focus/skip-focus.directive.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  Directive,
+  ElementRef,
+  Input,
+  OnChanges,
+  SimpleChanges,
+} from '@angular/core';
+import { WindowRef } from '@spartacus/core';
+
+export interface SkipFocusConfig {
+  isEnabled: boolean;
+  skipSelectors?: string[];
+}
+
+@Directive({
+  selector: '[cxSkipFocus]',
+})
+export class SkipFocusDirective implements OnChanges {
```

Comment:
I'm trying to think of something to encaptulate what this directive does, while not being confused with the other `TrapFocusDirective` which takes a different approach. I looked at that directive for a while seeing if there is some simpler solution we can leverage. But it does not look built to easily be adjusted to handle our need. So this new directive is justified IMO.

As for naming, I am thinking something similar that highlights our specific implementation (disabling tabindex on all other elements outside of trap). `TrapFocusByTabIndexDirective`? Its long but clear... Maybe there is something just as clear but shorter?